src/cast_core/src/CrudActor.cpp

```diff
@@ -1206,21 +1294,129 @@ struct CrudActor::PhaseConfig {
                                    id);
         };
 
-        operations = phaseContext.getPlural<std::unique_ptr<BaseOperation>>(
-            "Operation", "Operations", addOperation);
+        auto addState =
+            [&](const Node& node,
+                const std::unordered_map<std::string, int> states) -> std::unique_ptr<State> {
+            //           State myState;
+            auto stateName = node["Name"].to<std::string>();
+            // Skipping repeat for now
+            // operations
+            BOOST_LOG_TRIVIAL(debug) << "Adding state " << stateName;
+            auto stateOperations = node.getPlural<std::unique_ptr<BaseOperation>>(
+                "Operation", "Operations", addOperation);
+            BOOST_LOG_TRIVIAL(debug)
+                << "Got operations for state. Number of operations: " << stateOperations.size();
+            // Transitions
+            BOOST_LOG_TRIVIAL(debug) << "Got operations";
+            std::vector<double> weights;
+            std::vector<Transition> transitions;
+            for (auto [k, transitionYaml] : node["Transitions"]) {
+                weights.emplace_back(transitionYaml["Weight"].to<double>());
+                BOOST_LOG_TRIVIAL(debug)
+                    << "Next state name is " << transitionYaml["To"].to<std::string>();
+                transitions.emplace_back(Transition{
+                    states.at(transitionYaml["To"].to<std::string>()),
+                    Delay(transitionYaml["SleepBefore"], GeneratorArgs{phaseContext.rng(id), id})});
+            }
```

Comment:
echoing a previous comment - tying the weights and the transitions to the same struct/pair would make this slightly easier since you could put that logic into a ctor:

```
for (const auto&& [k, xion] : node["Transitions"]) {
  transitions.emplace_back<WeightedTransition>(xion, states, phaseContext, id);
}
```